README.rst

```diff
@@ -0,0 +1,806 @@
+  string
+  https://github.com/spdx/spdx-spec/blob/master/chapters/3-package-information.md
+
+- versioneye uses a type, name and version
+  https://github.com/versioneye/
+
+Solution
+========
+
+A `purl` or package URL is an attempt to standardize existing approaches to
+reliably identify and locate software packages.
+
+A `purl` is a URL string used to identify and locate a software package in a
+mostly universal and uniform way across programing languages, package managers,
+packaging conventions, tools, APIs and databases.
+
+Such a package URL is useful to reliably reference the same software package
+using a simple and expressive syntax and conventions based on familiar URLs.
+
+
+purl
+~~~~~
+
+`purl` stands for **package URL**.
+
+A `purl` is a URL composed of six components::
+
+    type:namespace/name@version?qualifiers#subpath
+
+Components are separated by a specific character for unambiguous parsing.
+
+The defintion for each components is:
+
+- **type**: the package "type" or package "protocol" such as maven, npm, nuget,
+  gem, pypi, etc. Required.
+- **namespace**: some name prefix such as a Maven groupid, a Docker image owner,
+  a GitHub user or organization. Optional and type-specific.
+- **name**: the name of the package. Required.
+- **version**: the version of the package. Optional.
+- **qualifiers**: extra qualifying data for a package such as an OS,
+  architecture, a distro, etc. Optional and type-specific.
+- **subpath**: extra subpath within a package, relative to the package root.
+  Optional.
+
+
+Components are designed such that they for a hierarchy from the most significant
+on the left left to the least significant components to the right.
+
+
+A `purl` must NOT contain a URL Authority i.e. there is no support for
+`username`, `password`, `host` and `port` components. A `namespace` segment may
+sometimes look like a `host` but its interpretation is specific to a `type`.
+
+
+Some `purl` examples
+~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    bitbucket:birkenfeld/pygments-main@244fd47e07d1014f0aed9c
+
+    deb:debian/curl@7.50.3-1?arch=i386&distro=jessie
+
+    docker:cassandra@sha256:244fd47e07d1004f0aed9c
+    docker:gcr.io/customer/dockerimage@sha256:244fd47e07d1004f0aed9c
+
+    gem:jruby-launcher@1.1.2?platform=java
+    gem:ruby-advisory-db-check@0.12.4
+
+    github:package-url/purl-spec@244fd47e07d1004f0aed9c
+
+    golang:google.golang.org/genproto#googleapis/api/annotations
+
+    maven:org.apache.xmlgraphics/batik-anim@1.9.1?packaging=sources
+    maven:org.apache.xmlgraphics/batik-anim@1.9.1?repository_url=repo.spring.io/release
+
+    npm:%40angular/animation@12.3.1
+    npm:foobar@12.3.1
+
+    nuget:EnterpriseLibrary.Common@6.0.1304
+
+    pypi:django@1.11.1
+
+    rpm:fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
+    rpm:opensuse/curl@7.56.1-1.1.?arch=i386&distro=opensuse-tumbleweed
+
+(NB: some checksums are truncated for brevity)
```

Comment:
I see a potential problem here in that this is going to result in a lot of new URL schemes, but URL schemes require a lot of review to get approved.  [See here](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml).  I suspect they're going to reject this spec on that basis alone.  The obvious way around it would be to change these to look something like:

```
purl:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
purl:github/package-url/purl-spec@244fd47e07d1004f0aed9c
...
etc.
```

Although honestly if this is going to be a thing, and it's always written in a URL, the `url` part of `purl` seems obvious.  Why not put the focus on the package?  e.g.:

```
pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
pkg:github/package-url/purl-spec@244fd47e07d1004f0aed9c
...
etc.
```

You'll note that `pkg:` isn't yet claimed 😄 .

Reply:
@tgamblin I moved this URL and scheme discussion to this ticket: https://github.com/package-url/purl-spec/issues/4